packages/ddp-server/livedata_server.js

```diff
@@ -2,6 +2,33 @@ DDPServer = {};
 
 var Fiber = Npm.require('fibers');
 
+// Publication strategies define how we threat data from published cursors on collection-level
+// This allows someone to:
+// - Make a trade-off between bandwidth and memory use
+// - Have special (non-mongo) collections like volatile message queues
+const publicationStrategies = {
```

Comment:
I would recommend a declarative approach here instead of imperative, adding fields like

```javascript

const publicationStrategies = {
  NO_MERGE_NO_HISTORY: {
    name: 'NO_MERGE_NO_HISTORY',
    useCollectionView: false,
    doAccountingForCollection: false,
  },
  NO_MERGE: {
    name: 'NO_MERGE',
    useCollectionView: false,
    doAccountingForCollection: true,
  },
  SERVER_MERGE: {
    name: 'SERVER_MERGE',
    useCollectionView: true,
    doAccountingForCollection: true,
  }
}
```

So in the future it would be easier to maintain this code without searching for if/elses using these types for specific decisions, also when adding new types the decisions will be clear for each one.

So these functions wouldn't be necessary:
`_useCollectionView`, `_doAccountingForCollection`

Reply:
I agree that's better. I'll try to rework this PR and maybe I can continue on my modernization PR

Reply:
@filipenevola I have made these changes. I did not include the field 'name' as nothing currently uses it.